evm/combine-signatures.js

```diff
@@ -0,0 +1,273 @@
+'use strict';
+
+const { ethers } = require('hardhat');
+const fs = require('fs');
+const path = require('path');
+const {
+    getDefaultProvider,
+    utils: { computePublicKey, keccak256, getAddress, arrayify, defaultAbiCoder, hashMessage, recoverAddress },
+    Contract,
+} = ethers;
+const { Command, Option } = require('commander');
+const {
+    mainProcessor,
+    printInfo,
+    printWalletInfo,
+    getGasOptions,
+    printError,
+    validateParameters,
+    getContractJSON,
+    getRandomBytes32,
+    getEVMAddresses,
+} = require('./utils');
+const { handleTx } = require('./its');
+const { getWallet } = require('./sign-utils');
+const { addBaseOptions } = require('./cli-utils');
+const IAxelarGateway = getContractJSON('IAxelarGateway');
+const IAxelarAuth = getContractJSON('IAxelarAuth');
+
+function readSignatures() {
+    const signaturesDir = path.join(__dirname, '../signatures');
+    const signatureFiles = fs.readdirSync(signaturesDir);
+    const signatures = [];
+
+    signatureFiles.forEach((file) => {
+        const filePath = path.join(signaturesDir, file);
+        const fileContent = fs.readFileSync(filePath, 'utf8');
+
+        try {
+            const signature = JSON.parse(fileContent);
+            signatures.push(signature);
+        } catch (error) {
+            printError(`Error parsing JSON in file ${file}`, error.message);
+        }
+    });
+
+    return signatures;
+}
+
+function getAddressFromPublicKey(publicKey) {
+    const uncompressedPublicKey = computePublicKey(publicKey, false);
+    const addressHash = keccak256(`0x${uncompressedPublicKey.slice(4)}`);
+
+    return getAddress('0x' + addressHash.slice(-40));
+}
+
+async function processCommand(config, chain, options) {
+    const { address, action, privateKey } = options;
+
+    const contracts = chain.contracts;
+
+    if (address) {
+        validateParameters({ isValidAddress: { address } });
+    }
+
+    const rpc = chain.rpc;
+    const provider = getDefaultProvider(rpc);
+
+    const wallet = await getWallet(privateKey, provider, options);
+    await printWalletInfo(wallet);
+
+    printInfo('Batch Action', action);
+
+    switch (action) {
+        case 'computeMessageHash': {
+            const { commandId, sourceChain, sourceAddress, contractAddress, payloadHash, sourceTxHash, sourceEventIndex } = options;
+
+            validateParameters({
+                isNonEmptyString: { sourceChain, sourceAddress },
+                isValidAddress: { contractAddress },
+                isKeccak256Hash: { commandId, payloadHash, sourceTxHash },
+                isValidNumber: { sourceEventIndex },
+            });
+
+            const chainId = chain.chainId;
+            const commandID = commandId || getRandomBytes32();
```

Comment:
start from the largest value and decrement until you get one that hasn't been executed yet perhaps